### ISSUE TITLE
Create configurable groups of lints

### ISSUE THREAD

**Issue by suaviloquence** [obi1kenobi/cargo-semver-checks#835]
### Describe your use case

With the addition of lint-level config, it would be helpful to have **lint groups** that you can configure many similar lints at once with one config entry.  For example, given a group of `suspicious` lints that are warn-by-default (to prevent false positives), some users want to make them all `deny` without having to list every single lint.

### Describe the solution you'd like

The ability to configure something like 
```toml
[package.metadata.cargo-semver-checks.lints]
suspicious = "deny"
```

(of course, both `level` and `required-update` can be configured, just like an individual lint)

### Alternatives, if applicable

We need to think about whether we want one group per lint (like in clippy) or multiple.  Having just one would be less complicated to implement, but if we had multiple, we could have (e.g.), a group for all the different kinds of `#[must_use]` lints as well as all minor-by-default lints (which tokio wants?).  We would also have to think about priority of configuring groups vs configuring lints.  Cargo `[lints]` has a `priority` key, for example.
